source/slang/glsl.meta.slang

```diff
@@ -2825,21 +2825,1988 @@ public uint rayQueryGetIntersectionTypeEXT(rayQueryEXT q, bool committed)
     return 0;
 }
 
+// TODO: implementation of built-in variables; proper tests; these are stubs
+// likley related to the following issue since GLSL adds new 
+// 'system' variables: https://github.com/shader-slang/slang/issues/411
+
+#define glsl_caps_shader_subgroup(param1) __glsl_extension(param1) [require(glsl)] 
+#define spirv_caps_shader_subgroup(param1) __spirv_version(param1) [require(spirv)] 
+
+//macros used to make implementation and modification reasonable under regex
+#define macroAnyVecGenericBuiltIn __generic<T : __BuiltinType, let N : int>
+#define macroAnyVecGenericLogical __generic<T : __BuiltinLogicalType, let N : int>
+#define macroAnyVecGenericArithmetic __generic<T : __BuiltinArithmeticType, let N : int>
+#define macroAnyVec vector<T, N>
+
+__generic<T : __BuiltinType>
+void shader_subgroup_preamble() {
+    // checks needed for shader_subgroup functions; __requireGLSLExtension does not work, the call fails to find the OP
+    __target_switch
+    {
+    case glsl:
+        if (__type_equals<T, half>()) __requireGLSLExtension("GL_EXT_shader_subgroup_extended_types_float16");
+        if (__type_equals<T, uint8_t>()) __requireGLSLExtension("GL_EXT_shader_subgroup_extended_types_int8");
+        if (__type_equals<T, uint16_t>()) __requireGLSLExtension("GL_EXT_shader_subgroup_extended_types_int16");
+        if (__type_equals<T, uint64_t>()) __requireGLSLExtension("GL_EXT_shader_subgroup_extended_types_int64");
+        return;
+    case spirv:
```

Comment:
There needs to be an inline spirv block here for the spirv backend.

Reply:
Is this still needed?

Reply:
for `case glsl`: __requireGLSLExtension is required

for `case: spirv`: we are required to output some spirv_asm (including empty block) to avoid crash (tested just now to be sure). The OpCapability code added can be removed since Slang handles the Int64/16/8/float16 capability under the hood

Reply:
Can we just `return`?

Reply:
we can use 'return' without any asm output